core/src/main/java/hudson/model/ViewJob.java

```diff
@@ -236,14 +236,23 @@ private boolean terminating() {
         @Override
         public void run() {
             while (!terminating()) {
+                String jobName = null;
                 try {
-                    getNext()._reload();
+                    var next = getNext();
+                    jobName = next.getFullName();
+                    next._reload();
+                    jobName = null;
                 } catch (InterruptedException e) {
                     // treat this as a death signal
                     return;
-                } catch (Throwable t) {
```

Comment:
⚠️

Reply:
Yeah. This class should just be moved to https://github.com/jenkinsci/external-monitor-job-plugin/blob/9fc4fed70ffac8b8c52288359c7854bffcb498c4/src/main/java/hudson/model/ExternalJob.java#L46 anyway.